lua/ui/notify/enhancementqueue.lua

```diff
@@ -125,37 +127,36 @@ function ModifyBuildablesForACU(originalBuildables, selection)
+            local factionCategory = categories[faction]
+            local supportFactories = newBuildableCategories - categories.SUPPORTFACTORY
+
+            ---@param hq UserUnit
+            for _, hq in hqs do
+                local supportCategory = categories.SUPPORTFACTORY * factionCategory
+
+                if EntityCategoryContains(categories.TECH3, hq) then
+                    supportCategory = supportCategory * (categories.TECH3 + categories.TECH2)
+                elseif EntityCategoryContains(categories.TECH2, hq) then
+                    supportCategory = supportCategory * categories.TECH2
+                end
 
-        -- LAND
-        UISelectionByCategory('LAND RESEARCH TECH3 ' .. faction, false, false, false, false)
-        if not GetSelectedUnits() then
-            newBuildableCategories = newBuildableCategories - (categories.LAND * categories.SUPPORTFACTORY * categories.TECH3 * factionCategory)
-            UISelectionByCategory('LAND RESEARCH TECH2 ' .. faction, false, false, false, false)
-            if not GetSelectedUnits() then
-                newBuildableCategories = newBuildableCategories - (categories.LAND * categories.SUPPORTFACTORY * categories.TECH2 * factionCategory)
-            end
-        end
-
-        -- AIR
-        UISelectionByCategory('AIR RESEARCH TECH3 ' .. faction, false, false, false, false)
-        if not GetSelectedUnits() then
-            newBuildableCategories = newBuildableCategories - (categories.AIR * categories.SUPPORTFACTORY * categories.TECH3 * factionCategory)
-            UISelectionByCategory('AIR RESEARCH TECH2 ' .. faction, false, false, false, false)
-            if not GetSelectedUnits() then
-                newBuildableCategories = newBuildableCategories - (categories.AIR * categories.SUPPORTFACTORY * categories.TECH2 * factionCategory)
-            end
-        end
-
-        -- Naval
-        UISelectionByCategory('NAVAL RESEARCH TECH3 ' .. faction, false, false, false, false)
-        if not GetSelectedUnits() then
-            newBuildableCategories = newBuildableCategories - (categories.NAVAL * categories.SUPPORTFACTORY * categories.TECH3 * factionCategory)
-            UISelectionByCategory('NAVAL RESEARCH TECH2 ' .. faction, false, false, false, false)
-            if not GetSelectedUnits() then
-                newBuildableCategories = newBuildableCategories - (categories.NAVAL * categories.SUPPORTFACTORY * categories.TECH2 * factionCategory)
+                if EntityCategoryContains(categories.LAND, hq) then
+                    supportCategory = supportCategory * categories.LAND
+                elseif EntityCategoryContains(categories.AIR, hq) then
+                    supportCategory = supportCategory * categories.AIR
+                elseif EntityCategoryContains(categories.NAVAL, hq) then
+                    supportCategory = supportCategory * categories.NAVAL
+                end
+                supportFactories = supportFactories + supportCategory
```

Comment:
This wouldn't fully work if an HQ unlocks multiple domains.
```suggestion
                if EntityCategoryContains(categories.LAND, hq) then
                    supportFactories = supportFactories + supportCategory * categories.LAND
                end
                if EntityCategoryContains(categories.AIR, hq) then
                    supportFactories = supportFactories + supportCategory * categories.AIR
                end
                if EntityCategoryContains(categories.NAVAL, hq) then
                    supportFactories = supportFactories + supportCategory * categories.NAVAL
                end
```